rust_src/src/windows.rs

```diff
@@ -622,4 +622,96 @@ pub extern "C" fn CURRENT_MODE_LINE_HEIGHT(mut window: LispWindowRef) -> i32 {
     window.current_mode_line_height()
 }
 
+/// Return a list of windows on FRAME, starting with WINDOW.
+/// FRAME nil or omitted means use the selected frame.
+/// WINDOW nil or omitted means use the window selected within FRAME.
+/// MINIBUF t means include the minibuffer window, even if it isn't active.
+/// MINIBUF nil or omitted means include the minibuffer window only
+/// if it's active.
+/// MINIBUF neither nil nor t means never include the minibuffer window.
+#[lisp_fn(min = "0")]
+pub fn window_list(
+    frame: Option<LispFrameRef>,
+    minibuf: LispObject,
+    window: Option<LispWindowRef>,
+) -> LispObject {
```

Comment:
If you wrote the following `if` as a `match` you can avoid the `is_none` and the `unwrap`.